app/javascript/crayons/MultiSelectAutocomplete/MultiSelectAutocomplete.jsx

```diff
@@ -0,0 +1,317 @@
+    listItem.innerText = selectedItem;
+    selectedItemsRef.current.appendChild(listItem);
+
+    exitEditState(nextInputValue);
+    setSelectedItems(newSelections);
+    // Clear the currently displayed suggestions & active index
+    setSuggestions([]);
+    setActiveDescendentIndex(null);
+    // Clear the text input
+    const { current: input } = inputRef;
+    input.value = nextInputValue;
+    focusInput && input.focus();
+  };
+
+  const deselectItem = (deselectedItem) => {
+    setSelectedItems(selectedItems.filter((item) => item !== deselectedItem));
+
+    // We also update the hidden selected items list, so removals are announced to screen reader users
+    selectedItemsRef.current.querySelectorAll('li').forEach((selectionNode) => {
+      if (selectionNode.innerText === deselectedItem) {
+        selectionNode.remove();
+      }
+    });
+  };
+
+  const allSelectedItemElements = selectedItems.map((item, index) => (
+    <li key={item} className="w-max">
+      <div role="group" aria-label={item} className="flex mr-1 mb-1 w-max">
+        <Button
+          variant="secondary"
+          className="c-autocomplete--multi__selected p-1 cursor-text"
+          aria-label={`Edit ${item}`}
+          onClick={() => enterEditState(item, index)}
+        >
+          {item}
+        </Button>
+        <Button
+          variant="secondary"
+          className="c-autocomplete--multi__selected p-1"
+          aria-label={`Remove ${item}`}
+          onClick={() => deselectItem(item)}
+        >
+          <Icon src={Close} />
+        </Button>
+      </div>
+    </li>
+  ));
+
+  // When a user edits a tag, we need to move the input inside the selected items
+  const splitSelectionsAt =
+    inputPosition !== null ? inputPosition : selectedItems.length;
+
+  const input = (
+    <li className="self-center">
+      <input
+        ref={inputRef}
+        autocomplete="off"
+        className="c-autocomplete--multi__input"
+        aria-activedescendant={
+          activeDescendentIndex !== null
+            ? suggestions[activeDescendentIndex]
+            : null
+        }
+        aria-autocomplete="list"
+        aria-labelledby="multi-select-label selected-items-list"
+        id="multi-select-combobox"
+        type="text"
+        onChange={handleInputChange}
+        onKeyDown={handleKeyDown}
+        onBlur={handleInputBlur}
+      />
+    </li>
+  );
+
+  return (
+    <Fragment>
+      <span
+        ref={inputSizerRef}
+        aria-hidden="true"
+        className="absolute pointer-events-none opacity-0 p-2"
+      />
+      <label id="multi-select-label">{labelText}</label>
+
+      {/* A visually hidden list provides confirmation messages to screen reader users as an item is selected or removed */}
+      <div className="screen-reader-only">
+        <p>Selected items:</p>
+        <ul
+          ref={selectedItemsRef}
+          className="screen-reader-only list-none"
+          aria-live="assertive"
```

Comment:
Refactor (blocking): Use the `role="alert"` attribute key/value pair instead.

```suggestion
          aria-live="assertive"
```

Reply:
Unfortunately `role="alert"` doesn't give us the extra control we get with `aria-relevant` - to announce only the added/removed text and avoid reading out the whole list every time something is changed.